sql/src/main/java/io/crate/operation/operator/EqOperator.java

```diff
@@ -63,12 +69,54 @@ public Boolean evaluate(Input[] args) {
         return left.equals(right);
     }
 
+    private static class ArrayEqOperator extends CmpOperator {
+
+        protected ArrayEqOperator(FunctionInfo info) {
+            super(info);
+        }
+
+        @Override
+        protected boolean compare(int comparisonResult) {
+            return comparisonResult == 0;
+        }
+
+        @Override
+        public Boolean evaluate(Input[] args) {
+            Object[] left = (Object[]) args[0].value();
+            if (left == null){
+                return null;
+            }
+            Object[] right = (Object[]) args[1].value();
+            if (right == null){
+                return null;
+            }
+            return Arrays.deepEquals(left, right);
+        }
+
+        @Override
+        public Symbol normalizeSymbol(Function symbol) {
+            Symbol left = symbol.arguments().get(0);
+            Symbol right = symbol.arguments().get(1);
+
+            if (containsNull(left, right)) {
+                return Literal.NULL;
```

Comment:
afaik this should result in a false literal, also maybe first check if both are literals?